h2o-algos/src/main/java/hex/modelselection/ModelSelectionUtils.java

```diff
@@ -210,6 +604,16 @@ public static int findBestR2Model(double lastBestR2, GLMModel[] bestR2Models) {
         return currBestR2 > lastBestR2 ? bestIndex : -1;
     }
     
+    public static int findBestMSEModel(double lastMinMSE, ModelSelection.SweepModel[] currModels) {
+/*        double currMinMSE = Stream.of(currModels).map(x->x._errorVariance).min(Comparator.naturalOrder()).get();
+        int bestIndex = Stream.of(currModels).map(x->x._errorVariance).collect(Collectors.toList()).indexOf(currMinMSE);
+        return currMinMSE < lastMinMSE ? bestIndex : -1;*/
+        int bestIndex = IntStream.range(0, currModels.length)
+                .reduce((i,j) ->currModels[i]._errorVariance <= currModels[j]._errorVariance? i : j).getAsInt();
+        double currMinMSE = currModels[bestIndex]._errorVariance;
```

Comment:
If I read the spreadsheet correctly, your version is faster, so let's switch back to it. Thank you for benchmarking both!